src/observer-position.js

```diff
@@ -4,31 +4,47 @@ const PositionObserver = ObserverInterface(function PositionObserver(opts = {})
   if (!(this instanceof PositionObserver)) {
     return new PositionObserver(...arguments)
   }
-
-  this.onTop = opts.onTop
-  this.onBottom = opts.onBottom
-  this.onMaximized = opts.onMaximized
-  this._wasTop = true
-  this._wasBottom = false
-
+  this.on = {
+    top: opts.onTop,
+    bottom: opts.onBottom,
+    left: opts.onLeft,
+    right: opts.onRight,
+    maximized: opts.onMaximized
+  }
+  this._was = {
+    top: true,
+    bottom: false,
+    left: true,
+    right: false
+  }
   const viewport = Observer.call(this, opts)
   this.check(viewport.getState())
 })
 
 PositionObserver.prototype.check = function(viewportState) {
-  const { onBottom, onTop, onMaximized, _wasTop, _wasBottom, container, offset, once } = this
-  const { scrollHeight } = container
-  const { height, y } = viewportState
-  const atTop = y - offset <= 0
-  const atBottom = scrollHeight > height && height + y + offset >= scrollHeight
+  const { on, _was, container, offset, once } = this
+  const { scrollHeight, scrollWidth } = container
+  const { dimension, position } = viewportState
+
+  const at = {
+    top: position.y - offset <= 0,
+    bottom: scrollHeight > dimension.height && dimension.height + position.y + offset >= scrollHeight,
```

Comment:
stash the dimensions in variables so they don't have to be referenced multiple times.  May make an impact with a lot of elements. Have to add benchmarks.